src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -527,3 +527,59 @@ tool_check_step!(Bootstrap { path: "src/bootstrap", default: false });
 // `run-make-support` will be built as part of suitable run-make compiletest test steps, but support
 // check to make it easier to work on.
 tool_check_step!(RunMakeSupport { path: "src/tools/run-make-support", default: false });
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) struct CoverageDump {}
```

Comment:
nit: Could just be a unit struct

```suggestion
pub(crate) struct CoverageDump;
```